app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb

```diff
@@ -0,0 +1,145 @@
+  rescue StandardError => err
+    _log.log_backtrace(err)
+    my_signal(minimize_indirect, :post_ansible_run, err.message, 'error')
+  end
+
+  def poll_ansible_tower_job_status(interval)
+    set_status('waiting for tower job to complete')
+
+    tower_job_status = ansible_job.raw_status
+    if tower_job_status.completed?
+      ansible_job.refresh_ems
+      log_stdout(tower_job_status)
+      if tower_job_status.succeeded?
+        my_signal(minimize_indirect, :post_ansible_run, job_finish_message, 'ok')
+      else
+        my_signal(minimize_indirect, :post_ansible_run, 'Ansible engine returned an error for the job', 'error')
+      end
+    else
+      interval = 60 if interval > 60
+      my_signal(false, :poll_ansible_tower_job_status, interval, :deliver_on => Time.now.utc + interval)
+    end
+  rescue StandardError => err
+    _log.log_backtrace(err)
+    my_signal(minimize_indirect, :post_ansible_run, err.message, 'error')
+  end
+
+  def post_ansible_run(message, status)
+    my_signal(true, :finish, message, status)
+  end
+
+  def ansible_job
+    job_class.find(options[:tower_job_id])
+  end
+
+  def set_status(message, status = "ok")
+    _log.info(message)
+    super
+  end
+
+  def job_template
+    @template ||= ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript.find(options[:ansible_template_id])
+  end
+
+  alias initializing dispatch_start
+  alias finish       process_finished
+  alias abort_job    process_abort
+  alias cancel       process_cancel
+  alias error        process_error
+
+  private
+
+  attr_writer :minimize_indirect
+
+  def load_transitions
+    self.state ||= 'initialize'
+
+    {
+      :initializing                  => {'initialize'       => 'waiting_to_start'},
+      :start                         => {'waiting_to_start' => 'running'},
+      :launch_ansible_tower_job      => {'running'          => 'ansible_job'},
+      :poll_ansible_tower_job_status => {'ansible_job'      => 'ansible_job'},
+      :post_ansible_run              => {'ansible_job'      => 'ansible_done'},
+      :finish                        => {'*'                => 'finished'},
+      :abort_job                     => {'*'                => 'aborting'},
+      :cancel                        => {'*'                => 'canceling'},
+      :error                         => {'*'                => '*'}
+    }
+  end
+
+  def my_signal(no_queue, action, *args, deliver_on: nil)
+    if no_queue
+      signal(action, *args)
+    else
+      queue_signal(action, *args, :deliver_on => deliver_on)
+    end
+  end
+
+  def queue_signal(*args, deliver_on: nil)
+    priority = options[:priority] || MiqQueue::NORMAL_PRIORITY
+
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :method_name => "signal",
+      :instance_id => id,
+      :priority    => priority,
+      :role        => 'ems_operations',
+      :zone        => zone || job_template.manager.my_zone,
```

Comment:
@mkanoor @bzwei Does this setting make sense?

Reply:
LTGM. Do you have any concern?

Reply:
I just added this line.